src/duk_api_debug.c

```diff
@@ -140,22 +142,53 @@ DUK_EXTERNAL void duk_debugger_cooperate(duk_context *ctx) {
 	DUK_UNREF(processed_messages);
 }
 
+DUK_EXTERNAL duk_bool_t duk_debugger_notify(duk_context *ctx, duk_idx_t nvalues) {
+	duk_hthread *thr;
+	duk_idx_t top, idx;
+	duk_bool_t ret = 0;
+
+	DUK_ASSERT_CTX_VALID(ctx);
+	thr = (duk_hthread *) ctx;
+	DUK_ASSERT(thr != NULL);
+	DUK_ASSERT(thr->heap != NULL);
+
+	if (DUK_HEAP_IS_DEBUGGER_ATTACHED(thr->heap)) {
+		top = duk_get_top(ctx);
+		if (top < nvalues) {
+			DUK_ERROR_API((duk_hthread *) ctx, "not enough stack values for notify");
+		}
+		duk_debug_write_notify(thr, DUK_DBG_CMD_APPNOTIFY);
+		for (idx = top - nvalues; idx < top; idx++) {
+			duk_tval *tv = duk_get_tval(ctx, idx);
+			duk_debug_write_tval(thr, tv);
+		}
+		duk_debug_write_eom(thr);
+		if (DUK_HEAP_IS_DEBUGGER_ATTACHED(thr->heap)) {
```

Comment:
Maybe a short comment here, e.g.:

    /* Return non-zero (true) if we have a good reason to believe the notify was
     * delivered; if we're still attached at least a transport error was not indicated
     * by the transport write callback.  This is not a 100% guarantee of course.
     */